source/h2_decoder.c

```diff
@@ -653,6 +653,10 @@ static int s_state_fn_prefix(struct aws_h2_decoder *decoder, struct aws_byte_cur
         frame->stream_id,
         frame->payload_len);
 
+    if (decoder->frame_in_progress.type == AWS_H2_FRAME_T_DATA) {
+        /* We invoke the on_data_begin here to report the whole payload size */
+        DECODER_CALL_VTABLE_STREAM_ARGS(decoder, on_data_begin, frame->payload_len);
+    }
```

Comment:
[trivial] Let's handle this like we handle other states that have a begin() callback. Declare a new state that calls this function. Rather than having the common prefix() function call this DATA-specific callback

Reply:
nevermind, we talked and the other way is too confusing to be worth it